src/main/java/com/github/jbellis/jvector/disk/OnDiskGraphIndex.java

```diff
@@ -60,25 +64,36 @@ public int maxEdgesPerNode() {
         return M;
     }
 
-    /** return an Graph that can be safely querried concurrently */
+    /** return a Graph that can be safely queried concurrently */
     public GraphIndex.View<T> getView()
     {
         return new OnDiskView(readerSupplier.get());
     }
 
+    // TODO: This is fake generic until the reading functionality uses T
     public class OnDiskView implements GraphIndex.View<T>, AutoCloseable
     {
         private final RandomAccessReader reader;
+        private final Map<Integer, T> vectorCache;
```

Comment:
Shouldn't need the cache here.  We're guaranteed to touch each node only once in the search (Searcher's visited set makes sure of that).  Searcher should ask for the vectors when it expands a new node (right before it asks for the neighbors).